delft3dworker/views.py

```diff
@@ -419,32 +438,98 @@ def publish_world(self, request, pk=None):
 
         return Response({'status': 'Published scene to world'})
 
+    @list_route(methods=["post"])  # denied after publish to world
+    def publish_world_all(self, request):
+        queryset = Scene.objects.filter(owner=self.request.user).filter(
+            suid__in=request.data.getlist('suid', []))
+
+        try:
+            for scene in queryset:
+                scene.publish_world(request.user)
+        except ValueError, e:
+            return Response(
+                {'status': e.message},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        return Response({'status': 'Published scenes to world'})
+
     @detail_route(methods=["get"])
     def export(self, request, pk=None):
-        scene = self.get_object()
+        # Alternatives to this implementation are:
+        # - django-zip-view (sets mimetype and content-disposition)
+        # - django-filebrowser (filtering and more elegant browsing)
+
+        # from:
+        # http://stackoverflow.com/questions/67454/serving-dynamically-generated-zip-archives-in-django
 
         options = self.request.query_params.getlist('options', [])
-        # What we will export, now ; separated (doesn't work), should be list
-        # as in https://delft3dgt-local:8000/api/v1/scenes/44/
-        # export/?options=export_images&options=export_input
+        if len(options) == 0:
+            return Response({'status': 'No export options given'},
+                status=status.HTTP_400_BAD_REQUEST)
 
-        if len(options) > 0:
-            stream, filename = scene.export(options)
+        scene = self.get_object()
 
-            resp = HttpResponse(
-                stream.getvalue(),
-                content_type="application/x-zip-compressed"
-            )
-            resp[
-                'Content-Disposition'] = 'attachment; filename={}'.format(
-                    filename
-            )
+        # The zip compressor
+        # Open BytesIO to grab in-memory ZIP contents
+        # (be explicit about bytes)
+        stream = io.BytesIO()
+        zf = zipfile.ZipFile(stream, "w", zipfile.ZIP_STORED, True)
+        files_added = scene.export(zf, options)
+        zf.close()
+
+        if not files_added:
+            return Response({'status': 'Empty zip file: selected files do not exist'},
+                status=status.HTTP_400_BAD_REQUEST)
+
+        resp = HttpResponse(
+            stream.getvalue(),
+            content_type="application/x-zip-compressed"
+        )
+        resp[
+            'Content-Disposition'] = 'attachment; filename={}'.format(
+                '{}.zip'.format(slugify(scene.name))
+        )
+
+        return resp
+
+    @list_route(methods=["get"])
```

Comment:
Correct. I intended this implementation to be a parallel changeover:

- making as little as possible changes on the front-end would mean less work
- lowering the chances of introducing a bug that would make downloading data completely impossible

I would suggest phasing out the old export function when the new has proven it's worthy of existence (dramatic music).